contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/webmvc/blueprint/AttributeRepresentationModelAssembler.java

```diff
@@ -0,0 +1,134 @@
+package com.contentgrid.spring.data.rest.webmvc.blueprint;
+
+import com.contentgrid.spring.data.rest.mapping.Property;
+import com.contentgrid.spring.data.rest.mapping.jackson.JacksonBasedProperty;
+import com.contentgrid.spring.data.rest.validation.AllowedValues;
+import com.contentgrid.spring.data.rest.webmvc.blueprint.AttributeRepresentationModel.SearchParamRepresentationModel;
+import com.contentgrid.spring.querydsl.mapping.CollectionFiltersMapping;
+import com.fasterxml.jackson.annotation.JsonValue;
+import jakarta.persistence.Embedded;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.data.rest.webmvc.RootResourceInformation;
+import org.springframework.data.util.TypeInformation;
+import org.springframework.hateoas.mediatype.MessageResolver;
+
+@RequiredArgsConstructor
+public class AttributeRepresentationModelAssembler {
+
+    private final CollectionFiltersMapping collectionFiltersMapping;
+    private final MessageResolver messageResolver;
+
+    public Optional<AttributeRepresentationModel> toModel(RootResourceInformation information, List<Property> properties) {
```

Comment:
Maybe give `properties` a different name, to make it clearer that it's about nesting, so a path to the current property. Maybe `pathToProperty`?